mllib/src/main/scala/org/apache/spark/mllib/stat/impl/MultivariateGaussian.scala

```diff
@@ -30,33 +31,72 @@ import org.apache.spark.mllib.util.MLUtils
  * @param mu The mean vector of the distribution
  * @param sigma The covariance matrix of the distribution
  */
-private[mllib] class MultivariateGaussian(
+class MultivariateGaussian private[mllib] (
     val mu: DBV[Double], 
     val sigma: DBM[Double]) extends Serializable {
 
+  if (sigma.cols != sigma.rows) {
```

Comment:
This is fine, but you can also use ```require``` for checking input arguments (since that throws ```IllegalArgumentException``` and is a bit shorter).